### PR TITLE
EREGCSC-352 First draft of search

### PR DESCRIPTION
**Pull request by whytheplatypus**
Implements a site wide search feature.

### CODE REVIEW

File: regulations/static/regulations/css/scss/partials/_sidebar_right.scss [hunk 1]
```diff
@@ -3,4 +3,18 @@
...
+    div {
+        padding-left: 2em;
+        h4 {
+            margin: 0;
+        }
+        p {
+            color: $secondary_text_color;
+            margin: 0;
+        }
+    }
```
Comment (whytheplatypus) [inline regulations/static/regulations/css/scss/partials/_sidebar_right.scss:19]: label these more explicitly

File: regulations/views/reader.py [hunk 2]
```diff
@@ -91,11 +93,23 @@ def get_view_links(self, context, toc):
...
         }
 
-    def build_toc_url(self, part, subpart, section, version):
-        return reverse('subpart_reader_view', args=(part, subpart, version))
-
 
 class SectionReaderView(ReaderView):
+    def get(self, request, *args, **kwargs):
+        if request.GET.get("section-only"):
+            return super().get(request, *args, **kwargs)
```
Comment (whytheplatypus) [inline regulations/views/reader.py]: Change behavior to redirect to subpart with _part_ view as the default

File: regulations/urls.py [hunk 3]
```diff
@@ -14,7 +15,11 @@
     path('', HomepageView.as_view(), name='homepage'),
     path('<numeric:part>/', RegulationLandingView.as_view(), name="regulation_landing_view"),
     path('<numeric:part>/<version:version>/', PartReaderView.as_view(), name='part_reader_view'),
-    path('<numeric:part>/<numeric:section>/<version:version>/', SectionReaderView.as_view(), name='section_reader_view'),
     path('<numeric:part>/<subpart:subpart>/<version:version>/', SubpartReaderView.as_view(), name="subpart_reader_view"),
+    path('<numeric:part>/<numeric:section>/<version:version>/', SectionReaderView.as_view(), name='section_reader_view'),
```
Comment (cgodwin1) [inline regulations/urls.py]: Should be able to remove these. `section_reader_view` is referenced within:

- content/regulations_landing_pages/landing_433.html
- regulations-site/regulations/views/goto.py
- regulations-site/regulations/views/reader.py
- regulations-site/regulations/tests/urls_test.py
- regulations-site/regulations/templates/regulations/recent-changes.html
- regulations-site/regulations/templatetags/layers.py
- regulations-site/regulations/generator/section_url.py

File: regulations/views/utils.py [hunk 4]
```diff
@@ -57,6 +57,10 @@ def find_subpart(section, toc, subpart=None):
             value = find_subpart(section, el['sub_toc'], '-'.join(el['index'][1:]))
             if value is not None:
                 return value
+        elif 'Subjgrp' in el['index'] and 'sub_toc' in el:
+            value = find_subpart(section, el['sub_toc'], subpart)
+            if value is not None:
+                return value
```
Comment (cgodwin1) [inline regulations/views/utils.py]: Could move `if value is not None:` clause outside the conditionals after line 63?

File: regulations/views/search.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    def get_data(self):
+        return self.accumulate_results([])
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        all_versions = versions.fetch_regulations_and_future_versions()
+        results = self.get_data()
+        context['results'] = list(get_current_results(results, all_versions))
+        return {**context, **self.request.GET.dict()}
```
Comment (cgodwin1) [inline regulations/views/search.py]: For consistency, perhaps `get_context_data` should be the first method of the class, as it's immediately obvious to a future reader what is happening

**Review by cgodwin1 — APPROVED** [review]

**Pull request merged**